libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -743,6 +746,19 @@ protected virtual void OnSetScopedServices(DialogContext dialogContext)
             if (Generator != null)
             {
                 dialogContext.Services.Set(this.Generator);
+                if (this.Generator is ResourceMultiLanguageGenerator generator)
+                {
+                    // Set generator key into state.
+                    dialogContext.State.SetValue(GeneratorIdKey, generator.ResourceId);
+
+                    // Set language policy key into state.
+                    var languagePolicy = generator.LanguagePolicy ??
+                                dialogContext.Services.Get<LanguagePolicy>() ??
+                                new LanguagePolicy();
+                    dialogContext.State.SetValue(LanguagePolicyKey, languagePolicy);
+
+                    dialogContext.RegisterTemplateFunctions(Generator);
+                }
```

Comment:
I don't think we should do it this way. 
What I would do is:
1. Make sure global LanguagePolicy is promoted into "turn" memory scope 
You can do by 
* Change UseLanguagePolicy() to 
      dialogManager.InitialTurnState.Add<LanguagePolicy>(policy);
      ObjectPath.SetPathValue(dialogManager.InitialTurnState, "turn.languagePolicy", policy);

* Change AdaptiveDialogBot.cs SetupTurnState() to look like this:
      turnContext.TurnState.Add(_languagePolicy);
       ObjectPath.SetPathValue(turnContext.TurnState, "turn.languagePolicy", _languagePolicy);

This makes the global default language policy available via "turn.languagePolicy" and the local dialog language policy is available via "dialogclass.generator.languagePolicy".  
Now your functions can access the language policy by using "(dialogclass.generator.languagePolicy || turn.languagePolicy)" 

Then I would register all of the LG files as functions with the language neutral definition
foo.lg.TEMPLATEX
Which uses language policy to bind to foo.en-us.lg, foo.en.lg foo.lg to bind TEMPLATEX

This allows all of the lg functions to be mounted in way which is namespaced and isolated from the rest of the system.

Thoughts?

Reply:
Thanks for the advices for the states about generatorId and policy. That would be very helpful, i will try that. 

In terms of registering functions, that's something i probably didn't communicate well about the expectation. My expectation is it should NOT namespaced and isolated from the rest the system. I agree that a better isolation looks good from a system design POV. But from user POV that isn't helping.

For example, if we do that, it will mean, as a user, i have to remember that 
* when doing send response\prompts, i don't have to use prefix
* when doing other actions, i have to use prefix
* when doing choiceInput's choices, i don't have to use prefix (by your previous PR) though defined in LG

Isn't that confusing to user?